### PR TITLE
List users who _would_ be emailed

### LINKED ISSUE #88: Add ability to list emails a message would go to for a given app.

### PR DESCRIPTION
**Pull request by mikehale**
As a developer who needs to email heroku customers, and wants to send out email to only a known subset of users to begin with... provide an endpoint which lists the users who _would_ be emailed when the `notify_app` call is used.

Fixes #88 

For now to limit scope, because I'm not sure who all has access to telex, I only allow the `logdrain-remediation` producer to list the users who would be emailed for an app.

### CODE REVIEW

File: spec/endpoints/producer_api/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        expect(body).to eq [{
+          "role" => "admin",
+          "user" => {
+            "email" => user.email,
+            "id" => user.id
+          }
+        }]
+      end
+
+      it "responsds 422 when an invalid id is used" do
```
Comment (mathias) [inline spec/endpoints/producer_api/users_spec.rb]: Small typo here.

File: spec/endpoints/producer_api/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        expect(last_response.status).to eq 200
+        body = MultiJson.decode(last_response.body)
+        expect(body).to eq [{
+          "role" => "admin",
+          "user" => {
+            "email" => user.email,
+            "id" => user.id
+          }
+        }]
+      end
```
Comment (mathias) [inline spec/endpoints/producer_api/users_spec.rb]: Should we also have a user that should not appear here, to verify the logic?

Comment (mikehale) [inline spec/endpoints/producer_api/users_spec.rb]: This isn't really testing what should be listed... that is in the specs for `UserFinder`, just that properly formatted json is returned. I setup an expectation on `AppUserFinder` with a canned response.

**Comment by mikehale** [comment]
I think telex is almost abandonware. Maybe we should just get a +1 from @heroku/security and call it a day.

**Comment by mwoods79** [comment]
LGTM 👍 

Gonna let this sit in staging for a while.

@mikehale this should really be two PR's one for upgrades and one for changes.

**Comment by mikehale** [comment]
FWIW, I tested I deployed this to a test app on the platform, and nothing jumped out at me error wise.

**Comment by cji** [comment]
I think something may have gone wrong with the deploy here - `telex.heroku.com` was just taken over by a bug bounty researcher.

**Pull request merged**